### PR TITLE
Phase2 RECO/Egamma RECO modifications v1

### PR DESCRIPTION
**Pull request by archiron**
#### PR description:

These changes are targeting the Phase-2 validation only and have no effect on the Run 3 validation. The updates made in the recent [PR33421](https://github.com/cms-sw/cmssw/pull/33421)) for the miniAOD are ported to the RECO validation, they allow to make on single set of validation plots out of two input GsfElectron collections (one for the barrel electrons, one for the HGCAL electrons). In the case of the Run 3 validation, the same collection is used for both barrel and endcaps.

The modifications are made in Validation/RecoEGamma/plugins folder :
ElectronMcSignalValidator.cc/h, 
ElectronMcFakeValidator.cc/h,
and into the Validation/RecoEGamma/python folder :
ElectronMcSignalValidator_gedGsfElectrons_cfi.py, 
ElectronMcFakePostValidator_cfi.py, 
ElectronMcFakeValidator_gedGsfElectrons_cfi.py, 
ElectronMcSignalValidatorPt1000_gedGsfElectrons_cfi.py. 

We also made some corrections into Validation/RecoEGamma/test folder :
ElectronMcSignalPostValidation_cfg.py, 
ElectronMcSignalValidation_gedGsfElectrons_cfg.py, 
ElectronMcSignalPostValidationMiniAOD_cfg.py, 
ElectronMcSignalValidationMiniAOD_cfg.py, 
electronValidationCheck_Env.py,
ElectronMcFakePostValidation_cfg.py,
ElectronMcFakeValidation_gedGsfElectrons_cfg.py,
ElectronMcSignalPostValidationPt1000_cfg.py,
ElectronMcSignalValidationPt1000_gedGsfElectrons_cfg.py,

here is the comparison for the CMSSW_12_0_0_pre3 release :
RECOstd_PoPtrueVsEta_pfxb RECOmod_PoPtrueVsEta_pfxb
Onto the left we have the actual comparison, and on the right we have the comparison made after the modifications.

#### PR validation:

compilation is OK, basics tests (scram b runtests or local tests) are OK too.
runTheMatrix.py -l limited -i all --ibeos tests are fine (19 15 14 9 4 1 1 1 1 tests passed, 19 3 0 0 0 0 0 0 0 failed).
Most of errors are "DAS_ERROR" or "open file error" (140.56, 136.88811 for example), but some ran correctly with our modifications (28234.0_TTbar_14TeV, 11634.911_TTbar_14TeV).
here is the run-log : [runall-report-step123-.LOG](https://github.com/cms-sw/cmssw/files/6753522/runall-report-step123-.LOG)

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

@beaudett

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

File: Validation/RecoEgamma/python/ElectronMcFakePostValidator_cfi.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 electronMcFakePostValidator = DQMEDHarvester("ElectronMcFakePostValidator",
 
-  Verbosity = cms.untracked.int32(0),
+  Verbosity = cms.untracked.int32(1),
```
Comment (jfernan2) [inline Validation/RecoEgamma/python/ElectronMcFakePostValidator_cfi.py:11]: This change is supposed to be printing the ME, from:
https://github.com/cms-sw/cmssw/blob/2db55fa80748eeb2ea48abd580a15631e3d7811b/DQMOffline/EGamma/src/ElectronDqmHarvesterBase.cc#L377-L378
It is not desired to flood the logs, I guess you forgot to disable it for debugging perhaps?
Thanks

Comment (archiron) [reply Validation/RecoEgamma/python/ElectronMcFakePostValidator_cfi.py:11]: heuh, yes. sorry.
do you want I make the change and repost the PR?

Comment (jfernan2) [reply Validation/RecoEgamma/python/ElectronMcFakePostValidator_cfi.py:11]: Yes please.
And if at the same time you can certify that changes seen are inline with what you expect:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-07-01-2300+78a26f/43793/dqm-histo-comparison-summary.html
I believe they are but small stats confuse me and I am not an expert on EGamma. Thank you

**Comment by sroychow** [comment]
@archiron a couple of suggestions 

- There are multiple instances of ```if (!readAOD_)``` - please put all the blocks together.
- Same for ```if (bestGsfElectron.isEE())``` and EB. 
  Also, these 2 conditions can come as if(EB){...} else if(EE){...} block. 
- In this [line](https://github.com/cms-sw/cmssw/blob/2db55fa80748eeb2ea48abd580a15631e3d7811b/Validation/RecoEgamma/plugins/ElectronMcFakeValidator.cc#L2623-L2624) Can't you divide first and then take the sqrt once?

**Comment by archiron** [comment]
After making several changes as requested, we found some inconsistencies in the treatment (such as in the "_all" histograms).
So we have to close the PR and work on it.

**Pull request closed without merging**